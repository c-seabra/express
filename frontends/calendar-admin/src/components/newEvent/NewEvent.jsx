import React from 'react'

const NewEvent = ({ starts_at }) => (
  <form className='form'>
    <label htmlFor='form.title'>title: </label><br />
    <input id='form.title' type='text' /><br />

    <label htmlFor='form.date'>begins at: </label><br />
    <input id='form.date' type='text' placeholder={starts_at} /><br />

    <label htmlFor='form.location'>location: </label><br />
    <input id='form.location' type='text' /><br />

    <label htmlFor='form.description'>description: </label><br />
    <input id='form.description' type='text' /><br />

  </form>
)

export default NewEvent
