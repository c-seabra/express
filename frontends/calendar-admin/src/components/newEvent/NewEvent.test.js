import React from 'react';
import renderer from 'react-test-renderer';

import NewEvent from './NewEvent';

describe('NewEvent', () => {
  test('renders', () => {
    const component = renderer.create(
      <NewEvent />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('with props renders with placeholder', () => {
    const component = renderer.create(
      <NewEvent starts_at='Wednesday' />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
