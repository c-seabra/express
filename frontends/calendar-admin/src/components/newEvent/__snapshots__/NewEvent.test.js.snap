// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewEvent renders 1`] = `
<form
  className="form"
>
  <label
    htmlFor="form.title"
  >
    title: 
  </label>
  <br />
  <input
    id="form.title"
    type="text"
  />
  <br />
  <label
    htmlFor="form.date"
  >
    begins at: 
  </label>
  <br />
  <input
    id="form.date"
    placeholder={undefined}
    type="text"
  />
  <br />
  <label
    htmlFor="form.location"
  >
    location: 
  </label>
  <br />
  <input
    id="form.location"
    type="text"
  />
  <br />
  <label
    htmlFor="form.description"
  >
    description: 
  </label>
  <br />
  <input
    id="form.description"
    type="text"
  />
  <br />
</form>
`;

exports[`NewEvent with props renders with placeholder 1`] = `
<form
  className="form"
>
  <label
    htmlFor="form.title"
  >
    title: 
  </label>
  <br />
  <input
    id="form.title"
    type="text"
  />
  <br />
  <label
    htmlFor="form.date"
  >
    begins at: 
  </label>
  <br />
  <input
    id="form.date"
    placeholder="Wednesday"
    type="text"
  />
  <br />
  <label
    htmlFor="form.location"
  >
    location: 
  </label>
  <br />
  <input
    id="form.location"
    type="text"
  />
  <br />
  <label
    htmlFor="form.description"
  >
    description: 
  </label>
  <br />
  <input
    id="form.description"
    type="text"
  />
  <br />
</form>
`;
